File: src/org/idea_sp/SourcePawnFileTypeFactory.java
package org.idea_sp;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;
import org.jetbrains.annotations.NotNull;

public class SourcePawnFileTypeFactory extends FileTypeFactory {
    @Override
    public void createFileTypes(@NotNull FileTypeConsumer fileTypeConsumer) {
        fileTypeConsumer.consume(SourcePawnFileType.INSTANCE, "sp");
        fileTypeConsumer.consume(SourcePawnFileType.INSTANCE, "inc");
    }
}

File: src/main/kotlin/org/idea_sp/SourcePawnFileTypeFactory.kt
package org.idea_sp

import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory

class SourcePawnFileTypeFactory : FileTypeFactory() {
    override fun createFileTypes(consumer: FileTypeConsumer) {
        consumer.consume(SourcePawnFileType, "sp");
        consumer.consume(SourcePawnFileType, "inc");
    }
}